homeassistant/components/aqara/base.py

```diff
@@ -0,0 +1,278 @@
+
+    def scale_value_back(self, value: float | int) -> int:
+        """Return raw value for scaled."""
+        return int(value * (10**self.scale))
+
+    def remap_value_to(
+        self,
+        value: float,
+        to_min: float | int = 0,
+        to_max: float | int = 255,
+        reverse: bool = False,
+    ) -> float:
+        """Remap a value from this range to a new range."""
+        return remap_value(value, self.min, self.max, to_min, to_max, reverse)
+
+    def remap_value_from(
+        self,
+        value: float,
+        from_min: float | int = 0,
+        from_max: float | int = 255,
+        reverse: bool = False,
+    ) -> float:
+        """Remap a value from its current range to this range."""
+        return remap_value(value, from_min, from_max, self.min, self.max, reverse)
+
+    @classmethod
+    def from_json(cls, data: str) -> IntegerTypeData:
+        """Load JSON string and return a IntegerTypeData object."""
+        return cls(**json.loads(data))
+
+
+class DeviceValueRange(SimpleNamespace):
+    """device's value range.
+
+    Attributes:
+        type(str): value's type, which may be Boolean, Integer, Enum, Json
+        values(dict): value range
+    """
+
+    type: str
+    values: str
+
+
+@dataclass
+class EnumTypeData:
+    """Enum Type Data."""
+
+    range: list[str]
+
+    @classmethod
+    def from_json(cls, data: str) -> EnumTypeData:
+        """Load JSON string and return a EnumTypeData object."""
+        return cls(**json.loads(data))
+
+
+class AqaraEntity(Entity):
+    """Aqara base device."""
+
+    _attr_should_poll = False
+
+    def __init__(self, point: AqaraPoint, device_manager: AqaraDeviceManager) -> None:
+        """Init AqaraHaEntity."""
+        self._attr_unique_id = f"Aqara.{point.id}"
+        self.point = point
+        self.device_manager = device_manager
+        device = self.get_aqara_device()
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, string_dot_to_underline(self.point.did))},
+            manufacturer="Aqara",
+            name=f"{device.device_name}({self.point.did})",
+            model=device.model,
+            suggested_area=f"{device.position_name}",
+        )
+
+    def get_aqara_device(self) -> AqaraDevice:
+        """Get the device of the point."""
+        return self.device_manager.get_device(self.point.did)
+
+    @property
+    def name(self) -> str | None:
+        """Return Aqara device name."""
+        if isinstance(self.point.name, str) and self.point.name != "":
+            return f"{self.point.name}({self.point.did})"
+        if hasattr(self, "entity_description"):
+            if self.entity_description.name is not None and isinstance(
+                self.entity_description.name, str
+            ):
+                return f"{self.entity_description.name}({self.point.did})"
+        return f"({self.point.id})"
+
+    @property
+    def device_info(self) -> DeviceInfo:
```

Comment:
This property is also set in the constructor? Which one is the right one?

Reply:
removed